src/js/Content/Features/Store/Search/Filters/DiscountPercentSearchFilter.js

```diff
@@ -0,0 +1,133 @@
+                        <input type="hidden" name="as-discount-percent">
+                    </div>
+                </div>`;
+    }
+
+    setup(params) {
+
+        this._minCount = document.querySelector(".js-discount-percent-lower");
+        this._maxCount = document.querySelector(".js-discount-percent-upper");
+
+        for (const input of document.querySelectorAll(".js-discount-percent-input")) {
+
+            input.addEventListener("change", () => {
+                this._apply();
+
+                const minVal = this._minCount.value;
+                const maxVal = this._maxCount.value;
+                let val = null;
+
+                if ((minVal && Number(minVal) !== 0) || maxVal) {
+                    val = `${minVal}-${maxVal}`;
+                }
+
+                this.value = val;
+            });
+
+            input.addEventListener("keydown", e => {
+                if (e.key === "Enter") {
+
+                    // Prevents unnecessary submitting of the advanced search form
+                    e.preventDefault();
+
+                    input.dispatchEvent(new Event("change"));
+                }
+            });
+        }
+
+        super.setup(params);
+    }
+
+    _setState(params) {
+
+        let lowerCountVal = "";
+        let upperCountVal = "";
+
+        if (params.has("as-discount-percent")) {
+
+            const val = params.get("as-discount-percent");
+            const match = val.match(/(^\d*)-(\d*)/);
+
+            this._value = val;
+
+            if (match) {
+                let [, lower, upper] = match;
+                lower = parseInt(lower);
+                upper = parseInt(upper);
+
+                if (!isNaN(lower)) {
+                    lowerCountVal = lower;
+                }
+                if (!isNaN(upper)) {
+                    upperCountVal = upper;
+                }
+            }
+        }
+
+        if (lowerCountVal !== this._minCount.value) {
+            this._minCount.value = lowerCountVal;
+        }
+        if (upperCountVal !== this._maxCount.value) {
+            this._maxCount.value = upperCountVal;
+        }
+    }
+
+    _addRowMetadata(rows = document.querySelectorAll(".search_result_row:not([data-as-discount-percent])")) {
+
+        for (const row of rows) {
+            const discountPercent = this.getDiscountPercent(row);
+            row.dataset.asDiscountPercent = discountPercent;
+        }
+    }
+
+    getDiscountPercent(row) {
+        let discount = 0;
+        const discountEl = row.querySelector(".search_discount");
+        
+        if (!discountEl || discountEl.innerText === "") {
+            return 0;
+        }
+
+        const discountText = discountEl.innerText.replace("-", "").replace("%", "");
+        discount = parseInt(discountText);
+        if (isNaN(discount)) {
+            discount = 0;
+        }
```

Comment:
I think you can just do this? Also this method is only used here, so you can prefix it with "_".

```suggestion
        const discountText = discountEl.innerText.match(/\d+/);
        if (discountText !== null) {
            discount = Number(discountText);
        }
```

Reply:
Done and done. Thanks for the feedback!